scripts/accounting/seed-expense-categories-from-csv.ts

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env node
+
+import '../../server/env';
+
+import fs from 'fs';
+
+import axios from 'axios';
+import { program } from 'commander';
+import { parse } from 'csv-parse/sync'; // eslint-disable-line
+
+import models, { Op, sequelize } from '../../server/models';
+
+const DRY_RUN = process.env.DRY_RUN !== 'false';
+
+const loadFileContents = async (filePathOrUrl: string): Promise<string> => {
+  if (filePathOrUrl.match(/^https?:\/\//)) {
+    const response = await axios.get(filePathOrUrl);
+    return response.data;
+  } else {
+    const content = await fs.promises.readFile(filePathOrUrl, 'utf8');
+    return content;
+  }
+};
+
+type RunOptions = {
+  verbose: boolean;
+};
+
+const run = async (hostSlug: string, csvPath: string, options: RunOptions) => {
+  // Load host
+  const host = await models.Collective.findOne({
+    where: { slug: hostSlug },
+    include: [{ association: 'accountingCategories', required: false }],
+  });
+
+  if (!host) {
+    throw new Error(`Host ${hostSlug} not found`);
+  }
+
+  // Load CSV content
+  const rawContent = await loadFileContents(csvPath);
+  const parsedContent = parse(rawContent, {
+    delimiter: ',',
+    columns: true,
+    skip_empty_lines: true, // eslint-disable-line camelcase
+  });
+
+  // Iterate over rows
+  for (const row of parsedContent) {
+    const shortTransactionGroup = row['Short Group ID'];
+    const expenseAccountingCategoryCode = row['EXP CODE'];
+    if (!shortTransactionGroup || !expenseAccountingCategoryCode) {
+      if (options.verbose) {
+        console.error(`Missing data for row ${JSON.stringify(row)}`);
+      }
+      continue;
+    }
+
+    const expense = await models.Expense.findOne({
+      include: [
+        {
+          model: models.Transaction,
+          attributes: [],
+          required: true,
+          where: {
+            type: 'DEBIT',
+            kind: 'EXPENSE',
+            HostCollectiveId: host.id,
+            [Op.and]: [
+              sequelize.where(sequelize.cast(sequelize.col('TransactionGroup'), 'text'), {
+                [Op.startsWith]: shortTransactionGroup,
```

Comment:
I think `shortTransactionGroup` can collide, it should be safer to use expense ID.

Reply:
I would have preferred to do so, also for performance reasons. But the document has no such column; this is the only strong link with Open Collective. Here's a small extract of the data:

|  |  |  |  |  | Trans #  | Type  | Date       | Short Group ID | Merchant Reference # | Vendor Name      | Collective |  Platform Tag    | Platform Memo                                              | Balance Sheet Memo  | Account                        | Split                 | Class | Debit  | Credit | Amount |  | Detail Category          | EXP CODE | 
|--|--|--|--|--|----------|-------|------------|----------------|----------------------|------------------|------------|------------------|------------------------------------------------------------|---------------------|--------------------------------|-----------------------|-------|--------|--------|--------|--|--------------------------|----------| 
|  |  |  |  |  | "15,661" | Check | 01/01/2022 | 90b5e417       |                      | DIGITALOCEAN.COM | Affect     |  web / software  | Virtual Card charge from DIGITALOCEAN.COM to Affect        | Affect              | 7030 · Receipt Expenses- Other | 1140 · OTHER CLEARING |       | 5.00   |        | 5.00   |  | Online Subscriptions     | 7333     | 
|  |  |  |  |  | "15,662" | Check | 01/01/2022 | c49c7886       |                      | HEB ONLINE       |            |  -               | Virtual Card charge from HEB ONLINE to Stop The Sweeps ATX | Stop The Sweeps ATX | 7030 · Receipt Expenses- Other | 1140 · OTHER CLEARING |       | 116.25 |        | 116.25 |  | Program Food & Groceries | 7312     |

Reply:
Yeah, I think it is OK since this is not a critical operation.